coopdevs/timeoverflow
# Fixed "Remember me" in Login

![giphy](https://user-images.githubusercontent.com/935744/41371376-08bd9f0e-6f42-11e8-9bcc-45f0234e5b28.gif)

Here I'm trying to make the "Remember me" option work, which will be the default option in the mobile app. 

Instead of expiring the session *the hard way*, I'm using the `timeoutable` devise module. I'm also configuring the `rememberable` module to remember sessions for 4 weeks. 

I also did a little cleanup on the `alert` bootstrap component, so, instead of using our own styled alert messages, I'm using bootstrap ones, which look exactly the same :D 

I also updated devise from 4.4.1 to 4.4.3 🚀 

![](http://g.recordit.co/PWetgHeSsW.gif)

# UPDATE

How to test: 

Change `config.remember_for` and `config.timeout_in` in your environment. In the gif, timeout_in was set to 5.seconds and remember_for to 10.seconds. With those values:

- If I DON'T check Remember me, whenever I stay more than 5 seconds without doing a request, I'll be kicked out. 
- if I check Remember me, I can use the app without doing login again for 10 seconds. 

In the gif you can also appreciate another case:

- When you are kicked out and you go to a public page (like offers or demands), you'll see a flash message saying that your session expired but you are not redirected to the login. 

- Instead, if you navigate to a page where you need to be looged in to see, whenever your session expires you'll be redirected to the login page and you'll see the flash message there as well.
